front/src/modules/conflict/utils.ts

```diff
@@ -95,78 +97,64 @@ export default function addTrainNamesToConflicts(
     trainMap.set(timetableItem.id, timetableItem);
   }
 
-  return conflicts.map((conflict) => ({
-    ...conflict,
-    trainNames: getConflictTrainNames(conflict, trainMap),
-    trainCategories: getConflictTrainCategories(conflict, trainMap),
-  }));
+  return conflicts.map((conflict) => {
+    const names = getConflictTrainNames(conflict, trainMap);
+    const categories = getConflictTrainCategories(conflict, trainMap);
+    return {
+      ...conflict,
+      trainsData: names.map((name, idx) => ({ name, category: categories[idx] ?? null })),
+    };
+  });
 }
 
-const isTrainMatch = (trainName: string, selectedTrainName: string) => {
-  if (trainName.startsWith(`${selectedTrainName} `) || trainName === selectedTrainName) {
-    return true;
-  }
-  const baseNumber = Number(selectedTrainName);
-  const occurrenceNumber = Number(trainName);
-
-  if (Number.isNaN(Number(selectedTrainName)) || Number.isNaN(Number(trainName))) {
-    return false;
-  }
-  return (occurrenceNumber - baseNumber) % 2 === 0 && occurrenceNumber > baseNumber;
-};
-
 export function isTrainInConflict(
   conflict: ConflictWithTrainNames,
-  selectedTrainName: string,
-  selectedTrainId: TrainId | undefined
+  selectedTrainId: TrainId
 ): boolean {
   if (!selectedTrainId) return false;
 
-  if (isPacedTrainId(selectedTrainId) || isIndexedOccurrenceId(selectedTrainId)) {
-    return conflict.trainNames.some((trainName) => isTrainMatch(trainName, selectedTrainName));
+  const selectedId = extractEditoastIdFromTrainId(selectedTrainId);
+
+  if (isOccurrenceId(selectedTrainId) || isIndexedOccurrenceId(selectedTrainId)) {
```

Comment:
`isOccurrenceId` tests for both indexed occurrences and added occurrences

```suggestion
  if (isOccurrenceId(selectedTrainId)) {
```

Reply:
Done